src/server/json_family.cc

```diff
@@ -326,8 +420,80 @@ OpResult<string> OpDoubleArithmetic(const OpArgs& op_args, string_view key, stri
   return output.as_string();
 }
 
+OpResult<long> OpDel(const OpArgs& op_args, string_view key, string_view path) {
+  long total_deletions = 0;
+  if (path.empty()) {
+    auto& db_slice = op_args.shard->db_slice();
+    auto fres = db_slice.FindExt(op_args.db_ind, key);
+    if (IsValid(fres.first)) {
+      total_deletions += long(db_slice.Del(op_args.db_ind, fres.first));
+    }
+
+    return total_deletions;
+  }
+
+  OpResult<json> result = GetJson(op_args, key);
+  if (!result) {
+    return total_deletions;
+  }
+
+  vector<string> deletion_items;
+  auto cb = [&](const string& path, json& val) { deletion_items.emplace_back(path); };
+
+  json j = result.value();
+  error_code ec = JsonReplace(j, path, cb);
+  if (ec) {
+    VLOG(1) << "Failed to evaulate expression on json with error: " << ec.message();
+    return total_deletions;
+  }
+
+  if (deletion_items.empty()) {
+    return total_deletions;
+  }
+
+  json patch(json_array_arg, {});
+  reverse(deletion_items.begin(), deletion_items.end());  // deletion should finish at root keys.
+  for (const auto& it : deletion_items) {
+    string pointer = ConvertToJsonPointer(it);
+    if (pointer.empty()) {
```

Comment:
does this `if` still happen?

Reply:
No, in case of an invalid JsonPath fatal log is printed and the program is exited.